spectator/registry.go

```diff
@@ -50,19 +51,18 @@ type RegistryInterface interface {
 	Close()
 }
 
-// Used to validate that Registry implements RegistryInterface at build time.
-var _ RegistryInterface = (*Registry)(nil)
+// Used to validate that spectatordRegistry implements Registry at build time.
+var _ Registry = (*spectatordRegistry)(nil)
 
-// Registry is the main entry point for creating meters and emitting metrics.
-type Registry struct {
-	config      *Config
-	writer      writer.Writer
+type spectatordRegistry struct {
+	config *Config
+	writer writer.Writer
 	logger      logger.Logger
 	loggerMutex sync.RWMutex
 }
 
 // NewRegistryConfiguredBy loads a new Config JSON file from disk at the path specified.
-func NewRegistryConfiguredBy(filePath string) (*Registry, error) {
+func NewRegistryConfiguredBy(filePath string) (Registry, error) {
```

Comment:
Hmm, if you updated this signature to take an optional logger argument and then assigned that to the config, it would obviate the need for a mutable logger on the Registry itself, further simplifying that interface.

```suggestion
func NewRegistryConfiguredBy(filePath string, logger Logger) (Registry, error) {
```

Reply:
Removed the option to create a Registry from a file-based config. Thanks for the recommendation 🙌

Reply:
Nice, this should let you shrink surface area of the registry interface too!